skimage/feature/tests/test_peak.py

```diff
@@ -61,7 +61,7 @@ def test_flat_peak(self):
         image = np.zeros((5, 5), dtype=np.uint8)
         image[1:3, 1:3] = 10
         peaks = peak.peak_local_max(image, min_distance=1)
-        assert len(peaks) == 4
+        assert len(peaks) == 1
```

Comment:
As above. I get concerned when tests change.

In fact, although many of the behaviour changes in this PR can be considered bug fixes, I would call this one an API change and that would require a deprecation cycle. Can it be easily restored?